.github/workflows/test.yml

```diff
@@ -7,8 +7,29 @@ on:
     - "v*"
 
 jobs:
+  validate:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - name: Get commits
+      id: commits
+      run: |
+        echo "MAIN=$(git show -s --format="%H" origin/main)" >> $GITHUB_OUTPUT
+        echo "TAG=$(git rev-list -n 1 ${GITHUB_REF#refs/*/})" >> $GITHUB_OUTPUT
+    - name: Print commits
+      run: |
+        echo "Main commit: ${{ steps.commits.outputs.MAIN }}"
+        echo "Tag commit: ${{ steps.commits.outputs.TAG }}"
+    - name: Compare commits
+      if: ${{ steps.commits.outputs.MAIN != steps.commits.outputs.TAG }}
+      uses: actions/github-script@d556feaca394842dc55e4734bf3bb9f685482fa0
```

Comment:
what's `d556feaca394842dc55e4734bf3bb9f685482fa0`?

Reply:
It's the commit hash of the current version—to mitigate risk regarding the possibility that a malicious actor overwrites the tag. I'll add the version in a comment for readability